tests/cpu/compare_with_acts_seeding.cpp

```diff
@@ -405,20 +358,24 @@ TEST_P(CompareWithActsSeedingTests, Run) {
     // the externalSpacePoint() method
 
     // Count the number of matching seeds
-    // and push_back seed into sorted_seedVector
-    std::vector<Acts::Seed<SpacePoint>> sorted_seedVector;
-    int n_seed_match = 0;
-    for (auto& seed : seeds) {
+    std::size_t n_matched_acts_seeds = 0u;
+    for (const auto& traccc_seed : traccc_seeds) {
+        // Try to find the same Acts seed.
         auto it = std::find_if(
-            seedVector.begin(), seedVector.end(), [&](auto acts_seed) {
-                auto traccc_spB = spacepoints_per_event.at(seed.spB_link);
-                auto traccc_spM = spacepoints_per_event.at(seed.spM_link);
-                auto traccc_spT = spacepoints_per_event.at(seed.spT_link);
-
-                auto& triplets = acts_seed.sp();
-                const SpacePoint* acts_spB = &triplets[0]->externalSpacePoint();
-                const SpacePoint* acts_spM = &triplets[1]->externalSpacePoint();
-                const SpacePoint* acts_spT = &triplets[2]->externalSpacePoint();
+            acts_seeds.begin(), acts_seeds.end(), [&](const auto& acts_seed) {
+                const traccc::spacepoint& traccc_spB =
+                    spacepoints_per_event.at(traccc_seed.spB_link);
+                const traccc::spacepoint& traccc_spM =
+                    spacepoints_per_event.at(traccc_seed.spM_link);
+                const traccc::spacepoint& traccc_spT =
+                    spacepoints_per_event.at(traccc_seed.spT_link);
+
+                const traccc::spacepoint& acts_spB =
+                    acts_seed.sp()[0]->externalSpacePoint();
+                const traccc::spacepoint& acts_spM =
+                    acts_seed.sp()[1]->externalSpacePoint();
+                const traccc::spacepoint& acts_spT =
+                    acts_seed.sp()[2]->externalSpacePoint();
```

Comment:
just a though: at this point ACTS has created Seeds of proxies to traccc space points, and proxies have an index (corresponding to the position of the original space point in the original collection). Meaning you can simply access the indexes and use them for the comparison.

So here you can actually compare the `traccc::seed::link_type` against the acts space point proxy `index()` instead of comparing the coordinates (which I believe is what the `==` does, right?)